### DISCUSSION TITLE
If a type is passed to mapped_column, is it possible to use type annotation?

### DISCUSSION THREAD

**Discussion by mcarans** [sqlalchemy/sqlalchemy#11748] [Usage Questions] [answered]
`update_date = mapped_column(DateTime, nullable=False)`

If a column is defined as above (in a dependent library), assuming I can use my own Base, is it possible to use a type annotation map to define the mapping from datetime to DateTime? If so, how?

I want to use a mapping like this:
```
class ConversionNoTZ(TypeDecorator):
    """Convert from/to datetime with timezone from database columns that don't
    use timezone"""

    impl = DateTime
    cache_ok = True

    def process_bind_param(self, value: Optional[datetime], _):
        if value is None:
            return value
        if value.tzinfo:
            value = value.astimezone(timezone.utc)

        return value.replace(tzinfo=None)

    def process_result_value(self, value, _):
        if value is None:
            return value
        if value.tzinfo is None:
            return value.replace(tzinfo=timezone.utc)

        return value.astimezone(timezone.utc)
        
class Base(DeclarativeBase):
    type_annotation_map = {
        datetime: ConversionNoTZ,
    }
```

The above works for `update_date: Mapped[datetime] = mapped_column(nullable=False)` but not `update_date = mapped_column(DateTime, nullable=False)`

**Comment by CaselIT** [comment, marked as answer]
Hi,

No, since the type is explicitly passed in the right side and that takes precedence compared with the annotation on the left side

**Reply by mcarans** [reply]
Thanks for your help!
